tests/reporters/test_custom_x_reporter.py

```diff
@@ -0,0 +1,344 @@
+    )
+except Exception:
+    account = ChainedAccount("fake")
+
+
+@pytest.fixture
+def mock_reporter_contract(tellorx_master_mock_contract, tellorx_oracle_mock_contract):
+    """mock custom reporter contract"""
+    return account_fake.deploy(
+        Reporter,
+        tellorx_master_mock_contract.address,
+        tellorx_oracle_mock_contract.address,
+        0,
+    )
+
+
+@pytest_asyncio.fixture(scope="function")
+async def custom_reporter(
+    rinkeby_test_cfg,
+    tellorx_master_mock_contract,
+    tellorx_oracle_mock_contract,
+    mock_reporter_contract,
+):
+    """Returns an instance of an IntervalReporter using
+    the ETH/USD median datafeed."""
+    async with TelliotCore(config=rinkeby_test_cfg) as core:
+        custom_contract = Contract(mock_reporter_contract.address, mock_reporter_contract.abi, core.endpoint, account)
+        custom_contract.connect()
+        tellorx = core.get_tellorx_contracts()
+        tellorx.master.address = tellorx_master_mock_contract.address
+        tellorx.oracle.address = tellorx_oracle_mock_contract.address
+        tellorx.master.connect()
+        tellorx.oracle.connect()
+        r = CustomXReporter(
+            custom_contract=custom_contract,
+            endpoint=core.config.get_endpoint(),
+            account=account,
+            master=tellorx.master,
+            oracle=tellorx.oracle,
+            datafeed=eth_usd_median_feed,
+            expected_profit="YOLO",
+            transaction_type=0,
+            gas_limit=400000,
+            max_fee=None,
+            priority_fee=None,
+            legacy_gas_price=None,
+            gas_price_speed="safeLow",
+            chain_id=core.config.main.chain_id,
+        )
+        # send eth from brownie address to reporter address for txn fees
+        accounts[1].transfer(account.address, "1 ether")
+        return r
+
+
+@pytest.mark.asyncio
+async def test_interval_reporter_submit_once(custom_reporter):
+    """Test reporting once to the TellorX through custom contract
+    with three retries."""
+    r = custom_reporter
+
+    # Sync reporter
+    r.datafeed = None
+
+    EXPECTED_ERRORS = {
+        "Current addess disputed. Switch address to continue reporting.",
+        "Current address is locked in dispute or for withdrawal.",
+        "Current address is in reporter lock.",
+        "Estimated profitability below threshold.",
+        "Estimated gas price is above maximum gas price.",
+        "Unable to retrieve updated datafeed value.",
+    }
+
+    oracle_address = r.custom_contract.address
+    tx_receipt, status = await r.report_once()
+    # Reporter submitted
+    if tx_receipt is not None and status.ok:
+        assert isinstance(tx_receipt, AttributeDict)
+        assert tx_receipt.to in oracle_address
+    # Reporter did not submit
+    else:
+        assert not tx_receipt
+        assert not status.ok
+        assert status.error in EXPECTED_ERRORS
+
+
+async def gas_price(speed="average"):
```

Comment:
Since you're reusing all these test functions from other reporter tests, let's move 'em to a utils file in /tests/reporters/ and import them into each individual test file from there. So the code is more DRY & maintainable